manager/integration/tests/test_upgrade.py

```diff
@@ -283,15 +335,64 @@ def test_upgrade(upgrade_longhorn_repo_url,
     pod['spec']['volumes'] = [create_pvc_spec(pvc_name)]
     create_and_wait_pod(core_api, pod)
 
-    volume1 = client.by_id_volume(volume_name)
-    volume1.attach(hostId=host_id)
-    volume1 = wait_for_volume_healthy(client, volume_name)
+    # Attach the volume
+    for v in volumes:
+        if v.name == vol_revision_enabled_name or \
+                v.name == vol_revision_disabled_name:
+            volume = client.by_id_volume(v.name)
+            volume.attach(hostId=host_id)
+            wait_for_volume_healthy(client, v.name)
+
+    # Verify volume's engine image has been upgraded
+    for v in volumes:
+        volume = client.by_id_volume(v.name)
+        engine = get_volume_engine(volume)
+        assert engine.engineImage == new_ei.image
+        assert engine.currentImage == new_ei.image
 
+    # Check All volumes data
     for sspod_info in statefulset_pod_info:
         resp = read_volume_data(core_api, sspod_info['pod_name'])
         assert resp == sspod_info['data']
 
     res_pod_md5sum = get_pod_data_md5sum(core_api, pod_name, pod_data_path)
     assert res_pod_md5sum == pod_md5sum
 
-    check_volume_data(volume1, volume1_data)
+    check_volume_data(vol_revision_enabled,
+                      vol_revision_enabled_data_after_sys_upgrade)
+    check_volume_data(vol_revision_disabled,
+                      vol_revision_disabled_data_after_sys_upgrade)
+    check_volume_data(vol_rebuild,
+                      vol_rebuild_data_after_sys_upgrade)
+
+    # Delete one healthy replica for vol_rebuild to trigger the rebuilding
+    delete_replica_on_test_node(client, vol_rebuild_name)
+    # Make sure vol_rebuild replica is deleted
+    replica_count = 2
+    vol_rebuild = wait_for_volume_replica_count(client, vol_rebuild_name,
+                                                replica_count)
+    # vol_rebuild will become degraded and start replica rebuilding
+    # Wait for replica rebuilding to complete
+    # Verify the vol_rebuild is still healthy
+    vol_rebuild = wait_for_volume_degraded(client, vol_rebuild_name)
+    assert vol_rebuild.robustness == "degraded"
+    vol_rebuild = wait_for_volume_healthy(client, vol_rebuild_name)
+    assert vol_rebuild.robustness == "healthy"
+    assert len(vol_rebuild.replicas) == 3
+
+
+# Need add this test case into test_upgrade()
```

Comment:
Just to understand why we need to add these lines when this is backported?

Reply:
Hi @khushboo-rancher :
    That's my typo, I should remove these lines.